### PR TITLE
feat: invite user api

### PR DESCRIPTION
**Pull request by yashmehrotra**

### CODE REVIEW

File: auth/controllers.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	}
+
+	link, err := k.createRecoveryLink(identity.Id)
+	if err != nil {
+		errMsg := []byte(fmt.Sprintf(`{"error": "%v", "message": "error creating recovery link"}`, err))
+		return c.JSONBlob(http.StatusInternalServerError, errMsg)
+	}
+
+	body := fmt.Sprintf(inviteUserTemplate, reqData.FirstName, link)
+	mail := utils.NewMail("team@flanksource.com", reqData.Email, "User Invite", body, "text/html")
```
Comment (moshloop) [inline auth/controllers.go:36]: Can we make this a command line argument ?

Comment (yashmehrotra) [reply auth/controllers.go:36]: The `from` part ?

File: auth/controllers.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	}
+
+	link, err := k.createRecoveryLink(identity.Id)
+	if err != nil {
+		errMsg := []byte(fmt.Sprintf(`{"error": "%v", "message": "error creating recovery link"}`, err))
+		return c.JSONBlob(http.StatusInternalServerError, errMsg)
+	}
+
+	body := fmt.Sprintf(inviteUserTemplate, reqData.FirstName, link)
+	mail := utils.NewMail(reqData.Email, "User Invite", body, "text/html")
```
Comment (moshloop) [inline auth/controllers.go:36]: Can we use a mail package? i.e. mail.New()

Comment (yashmehrotra) [reply auth/controllers.go]: Yeah, makes more sense. Also created a new type so we can do:
```golang
m := mail.New(args)
err := m.Send()
```

File: mail/mailer.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+	m := gomail.NewMessage()
+	m.SetHeader("From", FromAddress)
+	m.SetHeader("To", to)
+	m.SetHeader("Subject", subject)
+	m.SetBody(contentType, body)
+	return &Mail{message: m}
+}
+
+func (m Mail) Send() error {
+	host := os.Getenv("SMTP_HOST")
```
Comment (moshloop) [inline mail/mailer.go:26]: Can we map these through on n the helm charts ?

**Pull request merged**